[8.0.1xx] No .deps.json for net472 Container task

rainersigwald opened the pull request:
### Summary

Resolve CG alerts for Containers integration by removing a useless `.deps.json` file.

MSBuild only respects a .deps.json in the net8.0 engine, so there's no need to create it for a net472 task. It includes references that aren't part of the SDK, like the downlevel `System.Text.Json` that is provided by MSBuild/Visual Studio, so removing it can avoid false positives from dependency scanners.

Fixes #42279 (for 8.0.1xx)

### Customer Impact

Avoid spurious CG-scanner alerts related to references in MSBuild task.

### Regression?

No, longstanding behavior that is of interest because of a new vulnerability causing false-positive scan alerts.

### Testing

Existing repo testing.

### Risk

Low--the deleted file was not used at build time. No SDK outputs reference the updated `Microsoft.IO.Redist` assembly.

rainersigwald committed:
No .deps.json for net472 Container task

MSBuild only respects a .deps.json in the net8.0 engine, so there's no
need to create it for a net472 task. It includes references that aren't
part of the SDK, like the downlevel `System.Text.Json` that is provided
by MSBuild/Visual Studio, so removing it can avoid false positives from
dependency scanners.

marcpopMSFT commented:
@rainersigwald have you found that not generating the deps file is enough to satisfy CG? I know other scanners go off of the deps but I thought that CG looked at the downloaded files and I don't think this change will keep microsoft.io.redist from being downloaded.

marcpopMSFT commented:
dotnet/templating has a similar issue in main. I tried your fix from this PR and the 6.0.0 io.redist is still downloaded and still shows up in the project.assets.json so I think CG will still flag it.

rainersigwald commented:
Yeah, this is only to get scanner clean. To get CG clean . . . I guess SDK will also have to bump IO.Redist in your transitive pinned versions, since there's not a way to edit a reference out of MSBuild (we'll try in https://github.com/dotnet/msbuild/issues/10453 but not retroactively).

rainersigwald commented:
@marcpopMSFT do you want me to do that in this PR or a new one?

marcpopMSFT commented:
> @marcpopMSFT do you want me to do that in this PR or a new one?

Dang, I had been hoping this would be enough. If you want to include it, go ahead. I'll go try to fix it in the templating as well.

rainersigwald committed:
Pin higher Microsoft.IO.Redist

```
darc add-dependency --name Microsoft.IO.Redist --version 6.0.1 --repo https://github.com/dotnet/runtime --commit e77011b31a3e5c47d931248a64b47f9b2d47853d --type toolset
```

And manually add the `PackageVersion` to `Directory.Packages.props`.

rainersigwald commented:
Just tried to do the easy thing and unfortunately I think it might change outputs more than I expect; I think we'll need to pin the new version AND ALSO reference it in some strategic locations so that we can't in this repo accidentally take a dependency on a higher version. Working on that but I'd say we should think about merging this then doing that as a follow-up.

rainersigwald commented:
Ah, ok, I think that was a bit overzealous. Because nothing in this repo actually depends on M.IO.Redist (as measured by `refdump -s -a:Microsoft.IO.Redist` in `artifacts/` after a build), bumping the version is ok. Commit incoming.

Forgind commented:
@rainersigwald,

Branding updates were merged, so I think this is good to go?

marcpopMSFT commented:
Rainer is oof. Merging so we can get the branch to branch codeflow for this going.
